lib/tree/tree.go

```diff
@@ -0,0 +1,98 @@
+package tree
+
+import (
+	"github.com/dedis/cothority/lib/dbg"
+)
+
+/*
+The tree library holds all hosts and allows for different ways to
+retrieve the hosts - either in a n-ary tree, broadcast or just
+ a selection.
+*/
+
+// Tree holds the structure about the current tree
+type Tree struct {
+	// The top-most peer of this tree
+	Root *TreeEntry
+	// The hash-id of this tree - same for the whole tree
+	Hash []byte
+	// The hash of the peer-list
+	HashPL []byte
+	// The branching factor
+	BF int
```

Comment:
We may not want to impose a tree topology restriction with BF. The tree should be created the way the initiator wants it to be.